docs/source/etcd-architecture.rst

```diff
@@ -0,0 +1,238 @@
+- :ref:`calico-bgp-rr-component`, an optional BGP route reflector for higher
+  scale.
+
+The following sections break down each component in more detail.
+
+
+.. _calico-felix-component:
+
+Felix
+-----
+
+Felix is the most important component in the Calico network: without it, no
+network programming can be achieved. It is a daemon that runs on every machine
+that provides endpoints: in most cases that means on nodes that host containers
+or VMs.
+
+Depending on the specific orchestrator environment, Felix is responsible for
+some or all of the following tasks:
+
+Interface Creation and Management
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+In some environments, particularly containerized ones, Felix is responsible
+for creating and managing network interfaces to its endpoints. This usually
+involves creating and managing 'veth pairs': a pair of virtual Ethernet
+interfaces that behave as though they're connected together with a single
+Ethernet cable.
+
+In all cases, regardless of what kind of environment it runs in, Felix also
+programs some other information about interfaces into the kernel. In
+particular, it may enable `Proxy ARP`_, `Proxy NDP`_, and IP forwarding for
+interfaces that it manages.
+
+.. _Proxy ARP: http://en.wikipedia.org/wiki/Proxy_ARP
+.. _Proxy NDP: http://en.wikipedia.org/wiki/Neighbor_Discovery_Protocol
+
+Route Programming
+~~~~~~~~~~~~~~~~~
+
+Felix is responsible for programming routes to the endpoints on its host into
+the Linux kernel FIB. This ensures that packets destined for those endpoints
+that arrive on at the host are forwarded accordingly.
+
+ACL Programming
+~~~~~~~~~~~~~~~
+
+Felix is also responsible for programming ACLs into the Linux kernel. These
+ACLs are used to ensure that only valid traffic can be sent between
+endpoints, and ensure that endpoints are not capable of circumventing
+Calico's security measures. For more on this, see :doc:`security-model`.
+
+State Reporting
+~~~~~~~~~~~~~~~
+
+Felix is responsible for providing data about the health of the network. In
+particular, it reports errors and problems with configuring its host. This data
+is written into etcd, to make it visible to other components and operators of
+the network.
+
+
+.. _calico-orchestrator-plugin:
+
+Orchestrator Plugin
+-------------------
+
+Unlike Felix there is no single 'orchestrator plugin': instead, there are
+separate plugins for each major cloud orchestration platform (e.g. OpenStack,
+Kubernetes). The purpose of these plugins is to bind Calico more tightly into
+the orchestrator, allowing users to manage the Calico network just as they'd
+manage network tools that were built in to the orchestrator.
+
+A good example of an orchestrator plugin is the Calico Neutron ML2 mechanism
+driver. This component integrates with Neutron's ML2 plugin, and allows users
+to configure the Calico network by making Neutron API calls. This provides
+seamless integration with Neutron, and lowers the barrier of entry to using
+Calico.
```

Comment:
Delete "and lowers .... Calico".  The fact that we've written Calico lowers the barrier ...  The fact that we've written this doc lowers the barrier ...  In other words, we could say that after almost every sentence in our documentation - but it doesn't really add anything.